subsys/input/Kconfig

```diff
@@ -0,0 +1,70 @@
+# Copyright 2023 Google LLC
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig INPUT
+	bool "Input"
+	help
+	  Include input subsystem and drivers in the system config.
+
+if INPUT
+
+module = INPUT
+module-str = input
+source "subsys/logging/Kconfig.template.log_config"
+
+config INPUT_INIT_PRIORITY
+	int "Input subsystem and drivers init priority"
+	default 90
+	help
+	  Input subsystem and drivers initialization priority.
+
+choice INPUT_MODE
+	prompt "Input event processing mode"
+	default INPUT_MODE_THREAD
+
+config INPUT_MODE_SYNCHRONOUS
+	bool "Process input events synchronously"
+	help
+	  Input events callbacks are processed synchronously in the context of
+	  the code that is reporting the event.
+
+config INPUT_MODE_THREAD
+	bool "Process input events in a dedicated thread"
+	depends on MULTITHREADING
+	help
+	  Input events are added to a message queue and the callbacks are
+	  processed asynchronously in a dedicated thread.
+
+endchoice
+
+if INPUT_MODE_THREAD
+
+config INPUT_THREAD_PRIORITY_OVERRIDE
+	bool "Override default input thread priority"
+	help
+	  Option to change the default value of input thread priority.
```

Comment:
How come this boolean is needed? Why not just expose `CONFIG_INPUT_THREAD_PRIORITY` always?

Reply:
Copied from the shell module, haven't dug too much into it but this pattern is used a fair bit, I think because you can't default to `K_LOWEST_APPLICATION_THREAD_PRIO` in Kconfig.